app/main.py

```diff
@@ -1,2 +1,57 @@
+from typing import Any, Hashable
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.hash_table: list[None | Node] = [None] * self.capacity
+        self.load_factor = 2 / 3
+        self.threshold = int(self.capacity * self.load_factor)
+        self.size = 0
+
+    def _index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+        while (
+            self.hash_table[index] is not None
+            and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+        return index
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size == self.threshold:
+            self._resize()
+
+        index = self._index(key)
+
+        if self.hash_table[index] is None:
+            self.size += 1
+
+        self.hash_table[index] = Node(key, value)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._index(key)
+        if self.hash_table[index] is None:
+            raise KeyError("Not found")
+        return self.hash_table[index].value
+
+    def __len__(self) -> int:
+        return self.size
+
+    def _resize(self) -> None:
+        old_table = self.hash_table
+        self.size = 0
+        self.capacity *= 2
```

Comment:
When resizing, the capacity is doubled, which is a common approach. However, ensure that this resizing strategy aligns with your performance requirements, as it can lead to increased memory usage.